### ISSUE TITLE
ImportError when running on pypy

### ISSUE THREAD

**Issue by oscargus** [contourpy/contourpy#132]
I tried testing Matplotlib with pypy and got the following error:
```
ImportError: /opt/hostedtoolcache/PyPy/3.8.13/x64/lib/pypy3.8/site-packages/contourpy/_contourpy.pypy38-pp73-x86_64-linux-gnu.so: undefined symbol: PyExc_ImportError
```

https://github.com/matplotlib/matplotlib/pull/23207 although I will remove the pypy build rather soon as it took quite some time to install and run.

No idea if it is possible/reasonable to support it, but thought I'd report it anyway,

**Comment by ianthomas23** [comment]
Thanks for the report, I can reproduce this.

Pypy packages are mostly supported already, but evidently not as much as I thought. A `pip install` from source or a `conda install` from conda-forge packages work OK, but seemingly the pypi packages are faulty. It therefore shouldn't be too hard for me to fix.

If you wanted to continue with this before I have fixed it, you could always install contourpy from source in the mpl CI. Compilation time is negligible compared to mpl.

**Comment by oscargus** [comment]
Thanks for the reply!

I'll give up on it (for now at least) since building and running the tests took two hours using pypy (as opposed to the 20 minutes for the other tests...).

**Comment by ianthomas23** [comment]
@oscargus I have fixed a bug that occurred when using PyPy (#135) and added building and testing using PyPy 3.7 and 3.8 to the github actions (#136). I have tried PyPy on macOS and Windows and they build and pass the tests fine, but just as you observed they take so long to run that I am choosing not to.

I will need to do another release (v1.0.3) for these changes to make it through to the PyPI wheels, but there are a few other build-related improvements I need to make before that. It should hopefully be next week. So if you ever decide to return to trying Matplotlib and PyPy there should be no problems from ContourPy again!
